### ISSUE TITLE
Slaves not connecting to master

### ISSUE THREAD

**Issue by cakrit** [netdata/netdata#6946]
Hi!
Here is a new one:
Master stopped receiving stream data from half of slaves. And of course "I didn't touch anything!" (apart from memory mode = dbengine on Master) :)
Here are last logs from aforementioned v17 Slave
```
2019-09-25 12:09:19: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:19: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:19: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:19: netdata ERROR : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:24: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:24: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:24: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:24: netdata ERROR : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:29: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:29: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:29: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:29: netdata ERROR : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:34: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:34: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:34: netdata INFO  : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:34: netdata ERROR : STREAM_SENDER[wtdb1] : STREAM wtdb1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
```

And same from one of the missing v13 Slaves:
```
2019-09-25 12:09:22: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:22: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:22: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:22: netdata ERROR : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:27: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:27: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:27: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:27: netdata ERROR : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:32: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:32: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:32: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:32: netdata ERROR : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
2019-09-25 12:09:37: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: connecting...
2019-09-25 12:09:37: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: initializing communication...
2019-09-25 12:09:37: netdata INFO  : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: waiting response from remote netdata...
2019-09-25 12:09:37: netdata ERROR : STREAM_SENDER[wtapp1] : STREAM wtapp1 [send to _MASTER_IP_:19999]: server is not replying properly (is it a netdata?).
```

Then I restarted netdata v17 Master and list of available slaves changed. Some became available again, others became unavailable. 

Should I start a new issue?

Thanks!

_Originally posted by @noobiek in https://github.com/netdata/netdata/issues/6852#issuecomment-534935227_

**Comment by cakrit** [comment]
This is most likely related to #6826, perhaps even a duplicate. We need to dive deeper ASAP, it does seem like we've done something to break compatibility here and we need to figure out what it is. 
I remember us adding some more metadata that the slaves send to the master at some point, perhaps that was not done in a backwards-compatible way? Still though, doesn't make sense that half of the slaves have an issue. We'll probably need debug logs enabled on the master (give instructions to @noobiek) and possibly need to add more debug messages than the ones we have, if we can't figure out what's happening. Smells like a pointer/memory corruption issue, since it's not easily repeatable.

**Comment by thiagoftsm** [comment]
I will work on this today.

**Comment by thiagoftsm** [comment]
Our user updated the original issue reporting that looks like the problem was related with DBengine configuration, I asked him to do not close the issue at least until Monday for we confirm.

Now about part of the log that you brought here, I tested yesterday and today and I could receive the same error message, for all the cases that I received the message on the client side I had in the server side:

```
STREAM [receive from [192.168.0.23]:35660]: too busy to accept new streaming request. Will be allowed in 2 secs. (errno 11, Resource temporarily unavailable)
```

but, we deliver a message in the client that is the same for different behavior, the error message `server is not replying properly (is it a netdata?).` is also a common message when we have the SSL as force on the server and we try to communicate using plan text.

I am considering to change the default message on slave to simplify the user life and our life when read these messages.

**Comment by thiagoftsm** [comment]
Our user closed the original issue, his problem was fixed after he adjusts the database, but I used the issue to do a deep research in possible problems that we are having in our stream using each one of the ```memory mode```  available with Netdata. In these tests I always executed the following steps:

- I configured the master for a memory mode and I start it.
- After few seconds I started the slave in a complete different computer, this computer is almost 10 years old and it was using wireless behind an old access point.
- I let master and slave running during two hours without touch them.

I can group the problems found this way, I am grouping them, because they repeat in the time with different memory modes:

- Slave did not send data in a short interval of time, the master reached the end of the stream file and it closed the connection, this is the most common problem.
-  The Master tried to use the command SET in a chart without dimension  raising messages like this `requested a SET on chart 'cpu.cpu2_softirqs' of host 'athenas', without a dimension. Disabling it.`
- The Master tried to read an information from the stream that was not complete yet. When it tries to parse it could not identify the command and closed the stream.
- I set a master with SSL enabled and I did not activate the SSL on slave, the master rejects the connection and the slave gives the error "Is this a Netdata?"

On the other hand, when I set a stream between master and slave in a VM using the same computer, this means without any connection problem, I could not have any error. This last test ran almost 4 hours.

I also verified that some memory modes have a bigger probability to have the errors listed than others.
